### ISSUE TITLE
FallbackWarning: int64 position ID arithmetic falls back to CPU during torch.compile

### ISSUE THREAD

**Issue by Sanjeev900** [torch-spyre/torch-spyre#4317]
### What happened?

I noticed this while running ibm-granite/granite-embedding-278m-multilingual on Spyre through the vLLM Transformers backend with torch.compile.

The following warnings are printed during warmup and on subsequent inference calls:

```
FallbackWarning: conversion from torch.int64 to torch.float32 is falling back to cpu
FallbackWarning: conversion from torch.float32 to torch.int64 is falling back to cpu
```

I traced these warnings to create_position_ids_from_input_ids, which is used by the RoBERTa/XLM-RoBERTa embedding code.

The position ID generation uses int64 operations such as arange, masked_fill, and cumsum. These operations are not currently supported natively on Spyre, so the conversions fall back to CPU.

The model still runs and produces the expected output, so this is not a correctness issue.

The issue is that the fallback happens during execution. I see the warning on every inference call, which means the position ID calculation is going through CPU on the inference path.

### What did you expect to happen?

Ideally, the int64 operations required for position ID generation would be supported directly on Spyre.

If native int64 support is not possible for these operations at the moment, it would be useful if the fallback did not result in the same warning being emitted on every inference call.

A compile-time warning, or some other indication that this path is expected to execute on CPU, would make the behavior easier to understand.

### How can we reproduce it?

```
import warnings
import torch
from vllm import LLM

# Treat FallbackWarning as an error to make it visible
warnings.filterwarnings("error", category=UserWarning, message=".*FallbackWarning.*")

llm = LLM(
    model="ibm-granite/granite-embedding-278m-multilingual",
    runner="pooling",
    max_model_len=64,
    max_num_seqs=1,
    model_impl="transformers",
)
llm.embed(["Hello world."])
# FallbackWarning raised on every call from create_position_ids_from_input_ids
```

### Any environmental details we need to know?

Model: ibm-granite/granite-embedding-278m-multilingual
Backbone: XLM-RoBERTa
vLLM: v0.28.0
torch-spyre: e02b78b

### Relevant log output

```Shell

```
